## VSCode에서 gensim 설치 안되는 관계로 Google Colab에서 gensim 설치 후 실행하기


In [1]:
pip install gensim

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 27.8/27.8 MB 51.2 MB/s eta 0:00:00


In [ ]:
import nltk

nltk.download("punkt_tab")

[nltk_data] Downloading package punkt_tab to /root/nltk_data...
[nltk_data]   Unzipping tokenizers/punkt_tab.zip.


True

In [ ]:
from sklearn.datasets import fetch_20newsgroups

train_news = fetch_20newsgroups(
    subset="train", remove=("headers", "footers", "quotes"), random_state=156
)

X_train = train_news.data
y_train = train_news.target

print(X_train[0])



What I did NOT get with my drive (CD300i) is the System Install CD you
listed as #1.  Any ideas about how I can get one?  I bought my IIvx 8/120
from Direct Express in Chicago (no complaints at all -- good price & good
service).

BTW, I've heard that the System Install CD can be used to boot the mac;
however, my drive will NOT accept a CD caddy is the machine is off.  How can
you boot with it then?

--Dave



In [ ]:
from gensim.models import Word2Vec
import nltk

tokenized = [nltk.word_tokenize(doc.lower()) for doc in X_train]

model = Word2Vec(
    sentences=tokenized, vector_size=100, min_count=5, sg=1
)  # sg=1: Skip-gram
print("어휘 수:", len(model.wv))

어휘 수: 23296


- model 안에 단어가 23296개 존재함 (5개보다 적게 사용된 단어 제거)


In [ ]:
model.wv["man"]

array([ 0.1529336 ,  0.35238805, -0.32891765,  0.05485531,  0.70444983,
       -1.0707538 , -0.3628916 ,  0.14517896, -0.15427975, -0.23827852,
        0.20471078, -0.84518224,  0.09480604,  0.19506222,  0.21977736,
       -0.10795031,  0.5219877 , -0.04273057, -0.4109177 ,  0.03435652,
       -0.08793393,  0.2753845 , -0.5814486 ,  0.04520393, -0.17979884,
        0.1717303 , -0.1102583 , -0.19113162, -0.04340173,  0.17231724,
        0.23510817, -0.0308464 ,  0.19364521, -0.17542402,  0.13239683,
       -0.21352802,  0.29616764,  0.12178297, -0.00167706, -0.12517036,
       -0.6437528 ,  0.6193934 ,  0.21873017, -0.08632728,  0.22669618,
       -0.23605949, -0.04774183,  0.19737792,  0.05350444,  0.3129586 ,
        0.2264351 , -0.45603967,  0.0422057 , -0.0356354 , -0.08927505,
        0.14566076, -0.21990544, -0.11405129,  0.17064069, -0.1207635 ,
        0.22468022, -0.0630951 , -0.17411697,  0.07156193,  0.2032609 ,
        0.5229858 ,  0.45969117,  0.17003149, -0.19062679, -0.05

In [ ]:
# 'man'이라는 단어와 가장 유사한 단어 5개 출력
print(model.wv.most_similar("man", topn=5))

[('woman', 0.811141848564148), ('satan', 0.7053996324539185), ('himself', 0.6907495260238647), ('allah', 0.6852189302444458), ('murderer', 0.6824158430099487)]


In [ ]:
# 'mac'이라는 단어와 가장 유사한 단어 5개 출력
print(model.wv.most_similar("mac", topn=5))

[('ibm', 0.8117982745170593), ('macintosh', 0.7874471545219421), ('clone', 0.772752583026886), ('pc', 0.7645373940467834), ('centris', 0.7629437446594238)]
